Upgrade HF state for different mappings (#5472)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

* Added new function `qml.bravyi_kitaev` to map fermionic Hamiltonians to qubit Hamiltonians.
  [(#5390)](https://github.com/PennyLaneAI/pennylane/pull/5390)

>   ```python

Reviewer: Reminder to add a full changelog entry. (Sorry if this PR is WIP)

* `qml.transforms.hamiltonian_expand` can now handle multi-term observables with a constant offset.
  [(#5414)](https://github.com/PennyLaneAI/pennylane/pull/5414)

> * The `qml.qchem.hf_state` function can now be used for parity and Bravyi-Kitaev basis.

Reviewer 2: ```suggestion
* The `qml.qchem.hf_state` function is upgraded to be compatible parity and Bravyi-Kitaev basis.
```

> * The `qml.qchem.hf_state` function is upgraded to be compatible with parity and Bravyi-Kitaev basis.

Reviewer: ```suggestion
* The `qml.qchem.hf_state` function is upgraded to be compatible with the parity and Bravyi-Kitaev bases.
```

Author: I think it is supposed to be basis not bases.

Reviewer: :thinking: Not a big deal, but I'd be fine with:
- upgraded to be compatible with the parity basis and Bravyi-Kitaev basis
- upgraded to be compatible with the parity and Bravyi-Kitaev bases

In the latter, the `bases` refers to both `parity` and `Bravyi-Kitaev` so hence I think it should be plural.

Or are `parity` and `Bravyi-Kitaev` _the same basis_? In which case I agree with you.

Reviewer 2: @ddhawan11 Please use _bases_ as Tom originally suggested.

Reviewer 2: @trbromley It is changed to _bases_ as you suggested.
